### PR TITLE
Added anonymous commenting settings to molo-gem

### PR DESCRIPTION
**Pull request by sewagodimo**
Added the anonymous settings to molo commenting views. Insteading of defaulting to "anonymous" when non-staff users comment anonymously., the html now retrieves a value from settings.

### CODE REVIEW

**Comment by sewagodimo** [comment]
Hey @codiebeulaine @KaitCrawford @miltontony
 I added the tests so check if the anonymous commenting alias is being translated. I have also update the requirements for molo.commenting to retrieve the most recent version of molo.commenting. I then updated .travis.yml to include all the `./manage.py compilemessages' instruction so that the translations do not fail. Whoop whoop!

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ install:
   - pip install coveralls
 script:
   - flake8
+  - python manage.py makemessages -l es
```
Comment (codiebeulaine) [inline .travis.yml]: perhaps let's makemessages for all languages and not es

**Comment by codiebeulaine** [comment]
@sewagodimo we need to add tests here for the the customised anonymous alias, also let's have a chat with @saeedmb8 about the translations.

File: .travis.yml [hunk 2]
```diff
@@ -12,6 +12,8 @@ install:
   - pip install coveralls
 script:
   - flake8
+  - python manage.py makemessages
+  - python manage.py compilemessages
```
Comment (saeedmb8) [inline .travis.yml]: we shouldn't add these in travis file, the command makemessages should always be done locally, also we already compile the messages in the [docker file ](https://github.com/praekelt/molo-gem/blob/develop/Dockerfile#L7)

File: gem/tests/test_views.py [hunk 3]
```diff
@@ -431,6 +440,23 @@ def test_comment_filters(self):
 
         self.assertFalse(comment_form.is_valid())
 
+    def test_anonymous_comment_translation(self):
```
Comment (saeedmb8) [inline gem/tests/test_views.py]: we don't need to test for pootle translations as they often get changed. so I don't think this is necessarily

File: gem/templates/base/comments/comment.html [hunk 4]
```diff
@@ -14,10 +14,10 @@
           {% trans "Big Sister" %}
         {% endif %}
     {% elif node.user_name.lower == 'anonymous' %}
-      {{node.user_name}}
+      {% trans settings.commenting.CommentingSettings.commenting_anonymous %}
```
Comment (saeedmb8) [inline gem/templates/base/comments/comment.html]: we are not able to translate a value of variable that is not set in the template. For Pootle you would need the text that you want to be translated in the template. so the above syntax is not right.

**Pull request closed without merging**